src/main/scala/temple/DSL/semantics/MetadataParser.scala

```diff
@@ -39,13 +40,39 @@ class MetadataParser[T <: Metadata]() {
         constructor(argMap.getArg(argKey, argType))
       })
 
-  /** A shorthand for [[temple.DSL.semantics.MetadataParser#registerKeyword]] with the same `key` used for both the
-    * metadata name and its single argument */
+  /** A shorthand for
+    * [[temple.DSL.semantics.MetadataParser#registerKeyword(java.lang.String, temple.DSL.semantics.ArgType, scala.Option, scala.Function1)]]
+    * with the same `key` used for both the metadata name and its single argument */
   final protected def registerKeyword[A](key: String, argType: ArgType[A])(constructor: A => T): Unit =
     registerKeyword(key, key, argType)(constructor)
 
-  /** Perform parsing by looking up the relevant function and
-    * [[scala.collection.IterableOnceOps#foldRight(java.lang.Object, scala.Function2)]] supports */
+  /**
+    * Add a handler for a new type of metadata, with an optional argument.
+    *
+    * @param metaKey The name of the metadata item to add
+    * @param argKey The name of the single argument to the metadata. Note that there is also
+    *               [[temple.DSL.semantics.MetadataParser#registerOptionalKeyword(java.lang.String, java.lang.String, temple.DSL.semantics.ArgType, scala.Function1)]]
+    *               if this is the same as `metaKey`.
+    * @param argType The type of the field to expect
+    * @param constructor The function to turn an optional input of type [[ArgType]] into a value of type [[T]]
+    * @tparam A The underlying type of the field, inferred from `argType`
+    */
+  final protected def registerOptionalKeyword[A](metaKey: String, argKey: String, argType: ArgType[A])(
+    constructor: Option[A] => T,
+  ): Unit =
+    matchers += (metaKey -> { args =>
+        implicit val context: Context = Context(metaKey)
+        val argMap                    = parseParameters(argKey -> Some(NoArg))(args)
+        constructor(argMap.getOptionArg(argKey, argType))
+      })
+
+  /** A shorthand for
+    * [[temple.DSL.semantics.MetadataParser#registerOptionalKeyword(java.lang.String, temple.DSL.semantics.ArgType, scala.Function1)]]
+    * with the same `key` used for both the metadata name and its single argument */
+  final protected def registerOptionalKeyword[A](key: String, argType: ArgType[A])(constructor: Option[A] => T): Unit =
+    registerOptionalKeyword(key, key, argType)(constructor)
```

Comment:
This line has no coverage, I included it purely for consistency. Should I remove it?

Reply:
Yeah I think so, it's easy to add in the future if we need it